modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,516 @@
+            }
+        }
+        return isWebAppDeployed;
+    }
+
+    /**
+     * Check the given Carbon application is deployed correctly. This method is wait for 90 seconds
+     * for deployment of Carbon application and each 500 milliseconds  of wait it will check the
+     * deployment status.
+     *
+     * @param appName - Name of the application.
+     * @return boolean - true if the application is listed.
+     * @throws LazyLoadingTestException - Exception throws when creating WebAppAdminClient.
+     */
+    protected boolean isCarbonAppListed(String appName) throws LazyLoadingTestException {
+        int deploymentDelayInMilliseconds = 90 * 1000;
+        log.info("waiting " + deploymentDelayInMilliseconds + " millis for Carbon Application to list" + appName);
+        ApplicationAdminClient appAdminClient;
+        String[] appList;
+        boolean doLoop = true;
+        boolean isAppListed = false;
+        long time;
+        long startTime;
+
+        try {
+            appAdminClient = new ApplicationAdminClient(backendURL, sessionCookie);
+        } catch (AxisFault axisFault) {
+            String customErrorMessage = "AxisFault Exception  when creating WebAppAdminClient object. backend URL:" +
+                    backendURL + " Session Cookie: " + sessionCookie;
+            log.error(customErrorMessage, axisFault);
+            throw new LazyLoadingTestException(customErrorMessage, axisFault);
+        }
+
+        startTime = System.currentTimeMillis();
+        while (((time = (System.currentTimeMillis() - startTime)) < deploymentDelayInMilliseconds) && doLoop) {
+            //List all applications
+            try {
+                appList = appAdminClient.listAllApplications();
+            } catch (ApplicationAdminExceptionException applicationAdminExceptionException) {
+                String customErrorMessage = "ApplicationAdminExceptionException Exception when when calling " +
+                        "listAllApplications() methods in WebAppAdminClient";
+                log.error(customErrorMessage, applicationAdminExceptionException);
+                throw new LazyLoadingTestException(customErrorMessage, applicationAdminExceptionException);
+            } catch (RemoteException remoteException) {
+                String customErrorMessage =
+                        "RemoteException Exception when calling listAllApplications() methods in WebAppAdminClient";
+                log.error(customErrorMessage, remoteException);
+                throw new LazyLoadingTestException(customErrorMessage, remoteException);
+            }
+            //Find given application is available in app list. if found  return isAppListed = true or else with false
+            if (Arrays.asList(appList).contains(appName)) {
+                isAppListed = true;
+                log.info(appName + " Carbon Application is listed in" + time + " millis");
+                doLoop = false;
```

Comment:
Yes, we can  if so we have two exit points in our loop. in current implimentation, exit condition is get checked at one place